File: code/python/cp.0.py
#!/usr/bin/python
# Written by andrewt@cse.unsw.edu.au  for COMP2041
# Simple cp implementation using line by line I/O

import sys,os
if len(sys.argv) != 3:
    sys.stderr.write("Usage: %s <infile> <outfile>\n" % sys.argv[0])
    # or (Python3 only):
    # print("Usage:",  sys.argv[0], "<infile> <outfile>", file=sys.stderr)
    sys.exit(1)
outfile = open(sys.argv[2], 'w')
for line in open(sys.argv[1]):
    outfile.write(line)

File: code/perl_arrays/cp.0.pl
#!/usr/bin/perl -w
# Written by andrewt@cse.unsw.edu.au  for COMP2041
# Simple cp implementation using line by line I/O

die "Usage: $0 <infile> <outfile>\n" if @ARGV != 2;

$infile = shift @ARGV;
$outfile = shift @ARGV;

open my $in, '<', $infile or die "Cannot open $infile: $!";
open my $out, '>', $outfile or die "Cannot open $outfile: $!";

while ($line = <$in>) {
    print $out $line;
}

close $in;
close $out;
exit 0;
